docs/pages/blog/material-ui-v6-sneak-peek.md

```diff
@@ -0,0 +1,193 @@
+
+The new `extendTheme` lets you enable media [prefers-color-scheme](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme) in one line:
+
+<codeblock>
+
+```js Theme
+import { extendTheme, CssVarsProvider } from '@mui/material/styles';
+
+const theme = extendTheme({
+  strategy: 'media',
+});
+
+function App({ children }) {
+  return <CssVarsProvider theme={theme}>{children}</CssVarsProvider>;
+}
+```
+
+```css Result
+:root {
+  --palette-background-default: #fff;
+  /* ...other variables */
+}
+
+@media (prefers-color-scheme: dark) {
+  :root {
+    --palette-background-default: #121212;
+    /* ...other variables */
+  }
+}
+```
+
+</codeblock>
+
+If you have a custom selector for dark mode, for example using data attributes, you can specify a custom value:
+
+<codeblock>
+
+```js Theme
+import { extendTheme, CssVarsProvider } from '@mui/material/styles';
+
+const theme = extendTheme({
+  strategy: '[data-mode-%s]', // the %s will be replaced with the mode
+});
+
+function App({ children }) {
+  return <CssVarsProvider theme={theme}>{children}</CssVarsProvider>;
+}
+```
+
+```css Result
+:root {
+  --palette-background-default: #fff;
+  /* ...other variables */
+}
+
+:where([data-mode-dark]) {
+  --palette-background-default: #121212;
+  /* ...other variables */
+}
+```
+
+</codeblock>
+
+### Custom styles for specific modes
+
+The new API `theme.applyStyles()` has been added to unify all supporting styling solutions, including Emotion, Styled-components and Pigment CSS.
+
+We will provide a codemod that migrates the code like an example below:
+
+```diff
+  const StyledInput = styled(InputBase)(({ theme }) => ({
+    padding: 10,
+    width: '100%',
+-    borderBottom: `1px solid ${
+-      theme.palette.mode === 'dark' ? '#30363d' : '#eaecef'
+-    }`,
++    borderBottom: `1px solid #eaecef`,
++    ...theme.applyStyles('dark', {
++      borderBottom: '1px solid #30363d',
++    })
+    '& input': {
+      borderRadius: 4,
+-      backgroundColor: '#fff',
++     ...theme.applyStyles('dark', {
++       backgroundColor: '#0d1117',
++     })
+    },
+  }));
+```
+
+## Container Queries
+
+We added support for [container queries](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_containment/Container_queries) based on the existing `theme.breakpoints` API.
+
+This feature lets you define styles based on the width of the parent container instead of the viewport.
+
+```jsx
+const Component = styled('div')(({ theme }) => ({
+  display: 'flex',
+  flexDirection: 'column',
+  gap: theme.spacing(2),
+  [theme.containerQueries.up('sm')]: {
+    flexDirection: 'row',
+  },
+  [theme.containerQueries('sidebar').up('400px')]: {
+    flexDirection: 'row',
+  },
+}));
+```
+
+It also works with the `sx` prop:
+
+```jsx
+<Card
+  sx={{
+    '@sm': {
+      flexDirection: 'row',
+    },
+    '@400/sidebar': {
+      flexDirection: 'row',
+    },
+  }}
+/>
+```
+
+## Stabilized Grid v2
+
+## Revamping the free templates
```

Comment:
```suggestion
## Revamping the free templates

Explore the new and enhanced [Material UI free templates](https://mui.com/material-ui/getting-started/templates/) to see these amazing features in action. We've fully revamped the templates to provide the perfect starting point for your project, whether you're adding sleek styles or using the template's sections.

<!-- First demo  -->

### Custom syles

The new custom styles we've added allow you to dive into the best approaches for customization or simply copy and paste some stylish elements.

<!-- Second demo -->

Stay tuned, as many more updates are coming in the upcoming months! We'd love to hear your feedback on which templates or sections you'd like to see next.
```

### First demo

For the first demo, I thought on having something like a carousel, showcasing the new templates, with links for the preview and source code.

![image](https://github.com/mui/material-ui/assets/37222944/e2812982-763d-4779-be70-64b3cd553ee6)


👉 [Figma prototype](https://www.figma.com/design/4uv2dT18rXJPZBbrbpw61q/Marketing-site?node-id=10026-123&t=Jum9nq4wYDtxx1Cd-11)


### Second demo

For the second, I thought on having something like we had for color inversion on Joy UI marketing page:

![image](https://github.com/mui/material-ui/assets/37222944/fb8d58b2-6994-491b-8846-009d7cd30faf)

👉 [Figma prototype](https://www.figma.com/design/4uv2dT18rXJPZBbrbpw61q/Marketing-site?node-id=10026-124&t=Jum9nq4wYDtxx1Cd-11)

Reference:

<img width="400" alt="Screenshot 2024-07-09 at 09 50 03" src="https://github.com/mui/material-ui/assets/37222944/fa31b62b-2e84-447d-9a0b-0b3dd520441e">

Reply:
Got it!